### ISSUE TITLE
`use-implicit-booleaness-not-len [C1802]` should not be emitted for generators

### ISSUE THREAD

**Issue by dylwil3** [pylint-dev/pylint#10100]
### Bug description

```python
# tmp.py
if len((x for x in [1, 2, 3])): # this code emits a TypeError
    print("yay!")
```

### Command used

```shell
uvx pylint --disable=all --enable=C1802 tmp.py
```

### Pylint output

```python
************* Module tmp
tmp.py:1:3: C1802: Do not use `len(SEQUENCE)` without comparison to determine if a sequence is empty (use-implicit-booleaness-not-len)

------------------------------------------------------------------
Your code has been rated at 5.00/10 (previous run: 2.50/10, +2.50)
```

### Expected behavior

No lint emitted. This should be a different error - `len` is not defined for generators in the first place. Notice, for example, that no lint is emitted for length used as condition with other standard Python objects that do not admit lengths (e.g. lambda expressions or functions).

### Pylint version

```shell
pylint 3.3.1
astroid 3.3.5
Python 3.11.5 (main, Sep  8 2023, 16:31:47) [Clang 14.0.3 (clang-1403.0.22.14.1)]
```

**Comment by nedelcu-ioan** [comment]
Hi, I’d like to work on this issue!

I believe the problem can be resolved by removing the commented line in the following code snippet:
```python3
# file: pylint/checkers/refactoring/implicit_booleaness_checker.py:128
        generator_or_comprehension = (
            nodes.ListComp,
            nodes.SetComp,
            nodes.DictComp,
            # nodes.GeneratorExp,
        )
```

Additionally, the test should be updated to:
```python3
# file: pylint/tests/functional/u/use/use_implicit_booleaness_not_len.py:128
    assert len(u + 1 for u in [])  # Should be fine
```
Let me know if this approach aligns with the expected resolution, and I’ll proceed to submit a PR.

**Comment by Pierre-Sassoulas** [comment]
Thank you for proposing to work on this, I assigned you to the issue :) 

Not sure if the proposed fix will work, but the new test looks fine (and we can also use the example given in the original issue).
